feat(docs): Split docs into developer and network versions

critesjosh opened the pull request:
Splits the docs into developer and network operator sections. This allows for cleaner versioning between the two sets of docs and makes them easier to maintain (we don't have to update 4 sets of node version docs for udpates).

alejoamiras commented on .github/workflows/nightly-docs-release.yml:
```diff
@@ -71,7 +71,8 @@ jobs:
 
           # Check if we already have docs for this nightly version
-          DOCS_VERSION_DIR="docs/versioned_docs/version-$NIGHTLY_TAG"
+          # For multi-instance docs, check the developer-specific versioned docs directory
```
This would make it so only nightlies are cut in devloper world, right? (I'm ok with that, just double checking!)

critesjosh replied:
correct, no nightly node docs

alejoamiras commented on .github/workflows/nightly-docs-release.yml:
```diff
@@ -126,7 +127,7 @@ jobs:
           echo "Generated Aztec.nr API docs for: ${{ env.NIGHTLY_TAG }}"
 
-      - name: Create Aztec nightly docs version
+      - name: Create Aztec nightly docs version (Developer docs only)
```
By the looks of these changes, seems that it would be easy to also have a nightly for node docs?

critesjosh replied:
yea it would be easy. not sure if this is something that we want. wdyt @sklppy88?

alejoamiras commented:
It runs awesome on my computer; I still think we need to improve the landing page, but I'm happy to move forward as-is as long as we don't forget to improve the landing lol.

FYI: had to push a small fix. was not able to add it as a comment since the huge PR broke my macOS with 48 GB of RAM :)

critesjosh committed:
feat(docs): Split docs into developer and network versions

Splits the docs into developer and network operator sections. This allows for cleaner versioning between the two sets of docs and makes them easier to maintain (we don't have to update 4 sets of node version docs for udpates).
